src/main/java/com/google/cloud/spanner/pgadapter/statements/ExtendedQueryProtocolHandler.java

```diff
@@ -98,6 +110,16 @@ public BackendConnection getBackendConnection() {
     return backendConnection;
   }
 
+  public static Attributes getMetricAttributes(DatabaseId databaseId) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    if (databaseId != null) {
```

Comment:
`databaseId` should never be null (at least not for a real connection, maybe in some tests it is). Could we therefore change this into a more direct implementation, where:
1. `metricAttributes` are always created when `ExtendedQueryProtocolHandler` is created.
2. This method is not static, but an instance method, and always just returns the actual `metricAttributes` of this `ExtendedQueryProtocolHandler`. It should also not take `DatabaseId` (or anything else) as an input argument, as the `DatabaseId` is fixed for an `ExtendedQueryProtocolHandler`. The structure here is that:
1. Each connection from a client to PGAdapter has one `ConnectionHandler`.
2. There is a one-to-one relationship between `ConnectionHandler` and `ExtendedQueryProtocolHandler`.
3. There is a one-to-one relationship between `ExtendedQueryProtocolHandler` and `BackendConnection`.
4. All of the above classes also have exactly one `DatabaseId` that never changes during their lifetime.

Now it seems like `metricAttributes` sometimes needs to be created on-the-fly, but I don't think that is ever true.

Reply:
Yes, I was originally creating `metricAttributes` in `ExtendedQueryProtocolHandler` and pass it to the `BackendConnection`. But I thought we already pass `databaseId` into it in [here](https://github.com/GoogleCloudPlatform/pgadapter/blob/8f7dcb99bbce4334e5c4ffd56f7d3aaeaf78acc3/src/main/java/com/google/cloud/spanner/pgadapter/statements/ExtendedQueryProtocolHandler.java#L71). I need to construct the attributes in `Execute` of `BackendConnection`.

Reply:
I also think `databaseId` should not be null but I add this check due to an error. If I add `metricAttributes` as an instance method, can the `Execute` of `BackendConnection` access it? Let me explore it more.

Reply:
Please take a look my latest changes. Basically, I need the attributes in `Execute` of `BackendConnection`.